NOTICE-binary

```diff
@@ -73,6 +73,10 @@ Copyright 2005-2015 The Apache Software Foundation
 This product includes software developed at
 OW2 Consortium (http://asm.ow2.org/)
 
+This product includes software developed at
+NVIDIA (https://www.nvidia.com)
+* Copyright 2019 NVIDIA CORPORATION
+
```

Comment:
Ur, what does bring this `NOTICE-binary`?

Reply:
The lawyers at NVIDIA require that we place a copyright notice on contributions that are nontrivial. This is the first "significant" contribution to Apache Spark from NVIDIA so I am adding a copyright notice following the guidelines from ASF at https://www.apache.org/legal/src-headers.html#notice I followed the example of the lines right above it and it should be generic enough to cover any future contribution, so this is a one-time thing.

Reply:
Is there a precedent on this before? My understanding is that the software is developed at the Apache Software Foundation, not the contributor's company.

Reply:
I totally agree that this is not common.  I am not a lawyer and personally I agree with your interpretation, but the NVIDIA lawyers want the copyright notice in some form.  I can try and escalate it and see what has happened with NVIDIA for other ASF projects, but in the mean time if you could look at the rest of the patch that would be great.

Reply:
Will do. Thanks.

Reply:
Ping, @srowen . How do you think about this?

IIUC, since this PR is submitted directly to the ASF Apache Spark project, this cannot be a `NOTICE` for 3rd party work.
- https://www.apache.org/legal/src-headers.html#notice
- https://www.apache.org/legal/src-headers.html#3party
`The term "third-party work" refers to a work not submitted directly to the ASF by the copyright owner or owner's agent.`

Anyway, I understand that PMC can make a decision case-by-case.

Reply:
Yes, you would not be able to put this copyright message in an ASF project for code that was contributed to the project (vs code we include from third parties). It's not that it isn't true, it's that it's more specifically licensed to the ASF. We can ask for a CCLA if necessary but all the contributions would need to be under the AL2 license.

Reply:
Okay I have spoken with some of the legal team here and I have a better understanding of the issue, and an alternative solution. They are not really concerned about protecting NVIDIA by this, but instead want to make sure that there is clear traceability that this contribution is on behalf of NVIDIA and covered by their corporate CLA. From what I was told this is primarily to give our end users enough information that they can apply the Apache License properly especially in relation to the patent grant clause.

I am not a lawyer, this is just my understanding of what was said.  So I will be removing the change to this file, and instead I'll just make it clear in this PR, and any other PR I do going forward that is being contributed by on on behalf of my employer (NVIDIA).

Reply:
Thank you for removing this, @revans2 .